Replace submit queue with Tide in docs

nikhita opened the pull request:
Ref: https://groups.google.com/d/topic/kubernetes-dev/sp0dOVt4WsA/discussion

The submit queue is also mentioned in many places in `sig-scalability/` but I don't want to correct them in this PR since it touches their charters, etc.

/cc @spiffxp @cblecker @BenTheElder @guineveresaenger

nikhita commented on contributors/devel/automation.md:
```diff
@@ -6,9 +6,12 @@ Kubernetes uses a variety of automated tools in an attempt to relieve developers
-
 ## Submit Queue
 
+Submit Queue is deprecated. Kubernetes uses
```
Should this be removed completely or be marked as deprecated?

BenTheElder replied:
the submit queue doesn't even exist anymore (rm'ed) but people might have still heard of it.. I think we should probably keep this for a bit.

spiffxp replied:
I would change the heading to Tide, and this text can say something like "This project formerly used a Submit Queue, it has since been replaced by Tide"

nikhita commented on contributors/devel/e2e-tests.md:
```diff
@@ -612,11 +612,8 @@ A quick overview of how we run e2e CI on Kubernetes.
-queue](http://submit-queue.k8s.io/) on a subset of those tests if they fail (the
-subset is defined in the [munger config](https://git.k8s.io/test-infra/mungegithub/mungers/submit-queue.go)
-via the `jenkins-jobs` flag; note we also block on	`kubernetes-build` and
-`kubernetes-test-go` jobs for build and unit and integration tests).
```
Where is this defined now?

nikhita commented on contributors/devel/e2e-tests.md:
```diff
@@ -684,14 +681,11 @@ contend for resources; see above about [kinds of tests](#kinds_of_tests).
-quota,) they should remain with the `[Feature:.+]` label, and the suites that
-run them should be incorporated into the
-[munger config](https://git.k8s.io/test-infra/mungegithub/mungers/submit-queue.go)
-via the `jenkins-jobs` flag.
```
What is the equivalent alternative for this now?

BenTheElder commented:
/assign @cjwagner

BenTheElder commented on contributors/devel/automation.md:
```diff
@@ -48,6 +51,10 @@ If these tests pass a second time, the PR will be merged when this PR finishes r
 
 ## Github Munger
```
this probably doesn't need to be kept

BenTheElder commented on contributors/devel/e2e-tests.md:
```diff
@@ -576,7 +576,7 @@ suite, it receives a `[Feature:.+]` label, e.g. `[Feature:Performance]` or
 instead running in custom suites. If a feature is experimental or alpha and is
 not enabled by default due to being incomplete or potentially subject to
-breaking changes, it does *not* block the merge-queue, and thus should run in
+breaking changes, it does *not* block the merge pool, and thus should run in
```
this might be confusing without more background on pools

spiffxp replied:
It does _not_ block PR merges

spiffxp commented on contributors/devel/e2e-tests.md:
```diff
@@ -684,14 +681,11 @@ contend for resources; see above about [kinds of tests](#kinds_of_tests).
 Generally, a feature starts as `experimental`, and will be run in some suite
 owned by the team developing the feature. If a feature is in beta or GA, it
-*should* block the merge-queue. In moving from experimental to beta or GA, tests
+*should* block the merge pool. In moving from experimental to beta or GA, tests
```
block PR merges and releases

spiffxp commented on contributors/devel/e2e-tests.md:
```diff
@@ -612,11 +612,8 @@ A quick overview of how we run e2e CI on Kubernetes.
-via the `jenkins-jobs` flag; note we also block on	`kubernetes-build` and
-`kubernetes-test-go` jobs for build and unit and integration tests).
+continuous basis, and block merges via [Tide](https://git.k8s.io/test-infra/prow/cmd/tide)
+on a subset of those tests if they fail.
```
This whole section reads like it's left over from the old days of post-submits blocking the submit queue.  In that spirit, I would rewrite this to talk about release-blocking jobs, and reference https://testgrid.k8s.io/sig-release-master-blocking

spiffxp commented on contributors/devel/writing-good-e2e-tests.md:
```diff
@@ -10,7 +10,7 @@ designing, writing and debugging your end-to-end tests.  In
 intermittently for difficult-to-diagnose reasons are extremely costly
 in terms of blurring our regression signals and slowing down our
-automated merge queue.  Up-front time and effort designing your test
+automated merge pool.  Up-front time and effort designing your test
```
s/automated merge pool/merge velocity

spiffxp commented on contributors/guide/pull-requests.md:
```diff
@@ -72,7 +72,7 @@ Here's the process the pull request goes through on its way from submission to m
 1. If you're **not** a member of the Kubernetes organization, a Reviewer/Kubernetes Member checks that the pull request is safe to test. If so, they comment `/ok-to-test`. Pull requests by Kubernetes organization [members](/community-membership.md) do not need this step. Now the pull request is considered to be trusted, and the pre-submit tests will run:
 
-    1. Automatic tests run. See the current list of tests on the [MERGE REQUIREMENTS tab, at this link](https://submit-queue.k8s.io/#/info)
+    1. Automatic tests run. See the current list of tests on the [MERGE REQUIREMENTS tab, at this link](https://prow.k8s.io/tide#/info)
```
This link doesn't actually show tests that are run for every PR.  We're sort of lacking a clear list:
- The presubmits themselves are spread amongst a bunch of yaml files in https://github.com/kubernetes/test-infra/tree/master/config/jobs/kubernetes
- We could point to https://k8s-testgrid.appspot.com/presubmits-kubernetes-blocking
- We could point to https://prow.k8s.io/?repo=kubernetes%2Fkubernetes&type=presubmit

I guess the last link is my vote. But none of these clearly answer "how can I find out what these jobs do"

spiffxp commented on contributors/guide/pull-requests.md:
```diff
@@ -82,24 +82,26 @@ Here's the process the pull request goes through on its way from submission to m
-    1. Author has signed the CLA (`cncf-cla: yes` label added to pull request)
-    1. No changes made since last `lgtm` label applied
+1. The pull request enters the [merge pool](https://prow.k8s.io/tide)
+if the merge criteria is met. The [PR dashboard](https://prow.k8s.io/pr) shows
```
s/criteria is/criteria are/

spiffxp commented on contributors/guide/pull-requests.md:
```diff
@@ -82,24 +82,26 @@ Here's the process the pull request goes through on its way from submission to m
 ## Marking Unfinished Pull Requests
 
-If you want to solicit reviews before the implementation of your pull request is complete, you should hold your pull request to ensure that the merge queue does not pick it up and attempt to merge it. There are two methods to achieve this:
+If you want to solicit reviews before the implementation of your pull request is complete, you should hold your pull request to ensure that the merge pool does not pick it up and attempt to merge it. There are two methods to achieve this:
```
s/the merge pool/tide/

nikhita commented:
@BenTheElder @spiffxp Addressed comments. Ptal

Is it ok to do the re-write of https://github.com/kubernetes/community/blob/master/contributors/devel/e2e-tests.md as a follow up?

cjwagner commented on contributors/guide/pull-requests.md:
```diff
@@ -72,7 +72,7 @@ Here's the process the pull request goes through on its way from submission to m
 1. If you're **not** a member of the Kubernetes organization, a Reviewer/Kubernetes Member checks that the pull request is safe to test. If so, they comment `/ok-to-test`. Pull requests by Kubernetes organization [members](/community-membership.md) do not need this step. Now the pull request is considered to be trusted, and the pre-submit tests will run:
 
-    1. Automatic tests run. See the current list of tests on the [MERGE REQUIREMENTS tab, at this link](https://submit-queue.k8s.io/#/info)
+    1. Automatic tests run. See the current list of tests on the [MERGE REQUIREMENTS tab, at this link](https://prow.k8s.io/?repo=kubernetes%2Fkubernetes&type=presubmit)
```
There is no `MERGE REQUIREMENTS` tab at this link.

cjwagner commented on contributors/guide/pull-requests.md:
```diff
@@ -82,24 +82,26 @@ Here's the process the pull request goes through on its way from submission to m
 1. Follow the bot suggestions to assign an OWNER who will add the `/approve` label to the pull request
 
-Once the tests pass, all failures are commented as flakes, or the reviewer adds the labels `/lgtm` and `/approved`, the pull request enters the final merge queue. The merge queue is needed to make sure no incompatible changes have been introduced by other pull requests since the tests were last run on your pull request.
+Once the tests pass, all failures are commented as flakes, or the reviewer adds the labels `/lgtm` and `/approved`, the pull request enters the final merge pool. The merge pool is needed to make sure no incompatible changes have been introduced by other pull requests since the tests were last run on your pull request.
```
> all failures are commented as flakes

What does this mean?

nikhita replied:
> What does this mean?

Not sure, it was part of the old docs. I've updated it. Ptal.

nikhita commented on contributors/guide/owners.md:
```diff
@@ -209,21 +209,14 @@ is the state of today.
+  - responsible for determining **reviewers** and assigning to them
+- [Tide](https://git.k8s.io/test-infra/prow/cmd/tide):
+  - responsible for merging PRs
   - responsible for updating a GitHub status check explaining why a PR can't be merged (eg: a
```
Added.

cjwagner commented on contributors/devel/automation.md:
```diff
@@ -6,65 +6,26 @@ Kubernetes uses a variety of automated tools in an attempt to relieve developers
+#### Ready to merge status
 
 A PR is considered "ready for merging" by the submit queue if it matches the set
 of conditions listed in the [merge requirements tab](http://submit-queue.k8s.io/#/info)
```
Update this link to point to the tide dashboard. https://prow.k8s.io/tide

nikhita replied:
Done.

cjwagner commented on contributors/guide/contributor-cheatsheet.md:
```diff
@@ -15,7 +15,7 @@ A list of common resources when contributing to Kubernetes.
 
 - [Gubernator Dashboard - k8s.reviews](https://k8s-gubernator.appspot.com/pr)
-- [Submit Queue](https://submit-queue.k8s.io)
+- [Tide](https://prow.k8s.io/tide)
```
Please also remove reference to the cherrypick queue below.

nikhita replied:
Done.

nikhita commented:
I've created an issue https://github.com/kubernetes/community/issues/2759 for a follow-up update to the e2e tests doc.

spiffxp commented:
/approve
agree with the followup, will leave /lgtm for @cjwagner
